### PR TITLE
Dev/alha

### PR DESCRIPTION
**Pull request by alexjhawk**

### CODE REVIEW

File: src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/HistoricalDataManager.java [hunk 1]
```diff
@@ -348,11 +348,13 @@ public static ArrayList parseHistoricalFile(String filename)
         } catch (InterruptedException e) {
           e.printStackTrace();
         }
+      } else {
+        // Check for line with greater than 0 non-whitespace length
+        if (line.trim().length() > 0) {
```
Comment (it-hms) [inline src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/HistoricalDataManager.java:353]: the code checks for header here and sets the flag, but line is overwritten at the end of the while loop, and then headerEncoutered is checked again, but the line is different.

**Review by it-hms — CHANGES_REQUESTED** [review]

**Comment by it-hms** [comment]
@alexjhawk , perhaps the variable just needs to be renamed, but he code is hard to follow

**Comment by alexjhawk** [comment]
> @alexjhawk , perhaps the variable just needs to be renamed, but he code is hard to follow

Renamed to `headerPreviouslyEncountered`

File: src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/HistoricalDataManager.java [hunk 2]
```diff
@@ -322,15 +322,15 @@ public static ArrayList parseHistoricalFile(String filename)
     final BufferedReader reader = new BufferedReader(new FileReader(filename));
 
     // Create line counter and variable to store current line
-    int lineCount = 0;
+    boolean headerEncountered = false;
```
[the remark this answers is not in the document]
Comment (alexjhawk) [reply src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/HistoricalDataManager.java:325]: I don't know why we would remove the comment, but it is now updated.

File: src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/HistoricalDataManager.java [hunk 3]
```diff
@@ -322,15 +322,15 @@ public static ArrayList parseHistoricalFile(String filename)
...
     String line = reader.readLine();
 
     // Loop through lines in file until end and store data points
     ArrayList dataPoints = new ArrayList();
     while (line != null) {
 
       // Only parse lines 1 and greater, skip header
-      if (lineCount > 0) {
+      if (headerEncountered) {
```
[the remark this answers is not in the document]
Comment (alexjhawk) [reply src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/HistoricalDataManager.java:333]: Updated the comment

File: src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/HistoricalDataManager.java (hunk 1 above)
Comment (alexjhawk) [reply src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/HistoricalDataManager.java:353]: That is correct. We are checking that the header line has already been seen (encountered), and if it has, only then do we parse any line for data.

Comment (it-hms) [reply src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/HistoricalDataManager.java:353]: I missed the already part. It makes sense now.

**Review by it-hms — APPROVED** [review]

**Review by oliver-walker-hms — APPROVED** [review]
